client/java-spring-boot-autoconfigure/src/main/java/com/linecorp/centraldogma/client/spring/CentralDogmaSettings.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.client.spring;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.PositiveOrZero;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * Settings for a Central Dogma client.
+ *
+ * <h3>Example with {@code "host"} option</h3>
+ * <pre>{@code
+ * centraldogma:
+ *   hosts:
+ *   - replica1.examples.com:36462
+ *   - replica2.examples.com:36462
+ *   - replica3.examples.com:36462
+ *   accessToken: appToken-cffed349-d573-457f-8f74-4727ad9341ce
+ *   healthCheckIntervalMillis: 15000
+ *   useTls: false
+ * }</pre>
+ *
+ * <h3>Example with {@code "profile"} option</h3>
+ * <pre>{@code
+ * centraldogma:
+ *   profile: beta
+ *   accessToken: appToken-cffed349-d573-457f-8f74-4727ad9341ce
+ *   healthCheckIntervalMillis: 15000
+ *   useTls: false
+ * }</pre>
+ *
+ * <p>Note that {@code "healthCheckIntervalMillis"} and {@code "useTls"} are optional.</p>
+ */
+@ConfigurationProperties(prefix = "centraldogma")
+@Validated
+public class CentralDogmaSettings {
+    /**
+     * The Central Dogma client profile name. If not specified, {@code "hosts"} must be specified.
+     */
+    @Nullable
+    private String profile;
```

Comment:
From my use case, consider not only spring boot profile but also centraldogma profile seems confuse.

Spring boot supports including existing profiles, so may good to depend on framework feature.

Reply:
I agree that the name 'profile' may confuse some users, but it is indeed a 'profile', so let me add some note about this in the `client-java.rst` rather than renaming it.